client/src/components/Common/MapEditor/index.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import Geocode from "react-geocode";
+import styled from "styled-components";
+
+import { spacing32 } from "../../../resources/style-constants";
+import InputFormField from "../InputFormField";
+import Map from "../Map";
+
+const Container = styled.div`
+  width: 100%;
+  height: 30vw;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: flex-start;
+`;
+
+const MapContainer = styled.div`
+  height: 100%;
+  width: calc(60% - ${spacing32});
+  margin-right: ${spacing32};
+`;
+
+const OptionsContainer = styled.div`
+  height: 100%;
+  width: 40%;
+`;
+
+/**
+ * Displays a map with an edit field beside it. Use this
+ * to provide a GUI to update a maps geolocation (lat, lng).
+ * 
+ * IMPORTANT: For this api to work, you'll have to enable
+ * the geocoding API for your API key.
+ * 
+ * @label
```

Comment:
You should also check out PropTypes (there's an example of this in Common/Input/index.js). Using PropTypes you can specify the type of props (string, function, object, etc.) and which props are required for the component to run.

Reply:
I've been looking for a good way to type things ever since I started learning react.js 😆  This is super helpful, I'll do this!